Add exhaustive list of error codes and description of metric flag (#4422)
dapr/docs: daprdocs/content/en/operations/observability/metrics/metrics-overview.md

    enabled: false
```

## Configuring metrics for error codes

> You can enable additional metrics for [Dapr API error codes](https://github.com/dapr/dapr/blob/master/docs/reference/api/error_codes/) by setting `spec.metrics.recordErrorCodes` to `true`. See the specific metrics described in the [Dapr development docs](https://github.com/dapr/dapr/blob/master/docs/development/dapr-metrics.md).

Reviewer: I think this section should answer:
- why do I want to enable this?
- what happens when I enable this?

Also, mention metrics are collected and maybe add some metric examples

Reviewer 2: @jake-engelberg friendly bump

    enabled: false
```

## Configuring metrics for error codes

> You can enable additional metrics for [Dapr API error codes](https://github.com/dapr/dapr/blob/master/docs/reference/api/error_codes/) by setting `spec.metrics.recordErrorCodes` to `true`. Dapr APIs which communicate back to its caller may return standardized error codes. As described in the [Dapr development docs](https://github.com/dapr/dapr/blob/master/docs/development/dapr-metrics.md), a new metric called `error_code_total` will be recorded, which will allow monitoring of error codes triggered by application, code, and category. See [package errorcodes](https://github.com/dapr/dapr/blob/master/pkg/messages/errorcodes/errorcodes.go) for specific codes and categories.

Reviewer 2: ```suggestion
You can enable additional metrics for [Dapr API error codes](https://github.com/dapr/dapr/blob/master/docs/reference/api/error_codes/) by setting `spec.metrics.recordErrorCodes` to `true`. Dapr APIs which communicate back to their caller may return standardized error codes. As described in the [Dapr development docs](https://github.com/dapr/dapr/blob/master/docs/development/dapr-metrics.md), a new metric called `error_code_total` is recorded, which allows monitoring of error codes triggered by application, code, and category. See [the `errorcodes` package](https://github.com/dapr/dapr/blob/master/pkg/messages/errorcodes/errorcodes.go) for specific codes and categories.
```

Example configuration:
```yaml
apiVersion: dapr.io/v1alpha1
kind: Configuration
metadata:
  name: tracing
  namespace: default
spec:
  metrics:
    enabled: true

>     recordErrorCodes: true

Reviewer 3: You also need to;
1)  update this table in configuration setting https://docs.dapr.io/operations/configuration/configuration-overview/#metrics And in that table point to this article for the details. 
2) Update the configuration schema to add this under metrics section here https://docs.dapr.io/reference/resource-specs/configuration-schema/

Reviewer 3: @hhunter-ms  - Will do this. Do Jake you can ignore this comment.
